### PR TITLE
fix: defer load of google-p12-pem

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This should speed up start time a bit.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -50,6 +49,8 @@ class ErrorWithCode extends Error {
   }
 }
 
+let getPem: (filename: string) => Promise<string>;
```
Comment (jinwoo) [inline src/index.ts]: Should the type be `(...) -> Promise<string> | undefined`?

File: src/index.ts [hunk 2]
```diff
@@ -135,6 +136,9 @@ export class GoogleToken {
       }
       case 'application/x-pkcs12': {
         // *.p12 file
+        if (!getPem) {
+          getPem = (await import('google-p12-pem')).getPem;
```
Comment (jinwoo) [inline src/index.ts]: Is dynamic import already supported? I know TS supports it but I thought it worked only with `esnext`. How does this transpile to JS? Just `require('...')`?

Comment (ofrobots) [reply src/index.ts]: This change would need a common with the rationale for why deferred import is necessary.

File: src/index.ts (hunk 1 above)
Comment (JustinBeckwith) [reply src/index.ts]: I don't think so, this is the type of the getPem method from the module. Will double check :)

Comment (jinwoo) [reply src/index.ts]: I should've said `((...) -> Promise<string>) | undefined`.

In other words, `getPem` is initialized only after `google-p12-pem` is loaded. Before that, it is undefined and we need to make sure the type checker detects a programming error where `getPem` is referenced before it's initialized.

File: src/index.ts (hunk 2 above)
Comment (JustinBeckwith) [reply src/index.ts]: I'm as surprised as you  are that this worked in node.js 4 🤷‍♂️   It looks like it does melt down to a require:

```js
if (!!getPem) return [3 /*break*/, 7];
  return [4 /*yield*/, Promise.resolve().then(function () { return require('google-p12-pem'); })];
```

@ofrobots good call, added a comment.

File: src/index.ts (hunk 1 above)
Comment (JustinBeckwith) [reply src/index.ts]: Ahhh, I get it!  Great call.  I'm surprised that having an uninitialized declaration doesn't assume this in the compiler.  Updated 👍

**Comment by jinwoo** [comment]
It'd be great if you could write a test, as I commented earlier. But if it's too complicated, I'm fine with this.

**Pull request merged**